### PR TITLE
Asset Donor Search Backend

### PR DESCRIPTION
**Pull request by tilley14**

### CODE REVIEW

File: db/api/v1/db/assets.db.js [hunk 1]
```diff
@@ -22,25 +23,33 @@ const QUERY_FIND = `
...
+const find = async (sponsorId, projectId, assetType, donorCodes = undefined) => {
     let query = QUERY_FIND;
     let values = [];
-    if ((typeof sponsorId !== 'undefined') & (sponsorId > 0)) {
+    if ((typeof sponsorId !== 'undefined') && (sponsorId > 0)) {
         values.push(sponsorId);
         query = query + `AND sponsor_id = $${values.length}` + ' ';
     }
-    if ((typeof projectId !== 'undefined') & (projectId > 0)) {
+    if ((typeof projectId !== 'undefined') && (projectId > 0)) {
```
Comment (jedjas12) [inline db/api/v1/db/assets.db.js]: Formatting: Needs to be brought up to the closing brace

File: db/api/v1/db/assets.db.js [hunk 2]
```diff
@@ -22,25 +23,33 @@ const QUERY_FIND = `
...
         values.push(projectId);
         query = query + `AND project_id = $${values.length}` + ' ';
     }
-    if ((typeof assetType !== 'undefined') & (assetType > 0)) {
+    if ((typeof assetType !== 'undefined') && (assetType > 0)) {
```
Comment (jedjas12) [inline db/api/v1/db/assets.db.js]: Brace formatting

File: db/api/v1/db/assets.db.js [hunk 3]
```diff
@@ -22,25 +23,33 @@ const QUERY_FIND = `
...
         values.push(assetType);
         query = query + `AND asset_type_id = $${values.length}` + ' ';
     }
+    if ((typeof donorCodes !== 'undefined') && (donorCodes.length > 0)) {
```
Comment (jedjas12) [inline db/api/v1/db/assets.db.js]: Brace formatting

File: db/api/v1/db/assets.db.js (hunk 1 above)
Comment (tilley14) [reply db/api/v1/db/assets.db.js]: This is not an else if

Comment (APalumbo97) [reply db/api/v1/db/assets.db.js]: Not for a list of `if`s

File: db/api/v1/db/assets.db.js (hunk 2 above)
Comment (APalumbo97) [reply db/api/v1/db/assets.db.js]: Disagree

File: db/api/v1/db/assets.db.js (hunk 3 above)
Comment (APalumbo97) [reply db/api/v1/db/assets.db.js]: Disagree

File: db/api/v1/routes/validate.js [hunk 4]
```diff
@@ -509,6 +509,30 @@ const parseProject = (project) => {
...
+ * 
+ * @param {*} donor_codes - the donor code string or an array of donor codes
+ * @returns {ParseResult} a successful ParseResult if its a valid donor code
+ */
+const parseDonorCode = (donor_codes) => {
+    if (parseString(donor_codes)) {
+        return ParseResult.success([donor_codes]);
+    } else if (Array.isArray(donor_codes) && donor_codes.length != 0) {
+        var allStrings = true;
+        for(const code in donor_codes) {
```
Comment (APalumbo97) [inline db/api/v1/routes/validate.js]: Add a space between `for` and the `(`

File: db/api/v1/routes/validate.js [hunk 5]
```diff
@@ -509,6 +509,30 @@ const parseProject = (project) => {
...
+            if (!parseString(code)) { 
+                allStrings = false;
+                break;
+            }
+        }
+        if (allStrings)
```
Comment (APalumbo97) [inline db/api/v1/routes/validate.js]: Wrap this statement in a code block

**Pull request merged**